curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method.md

```diff
@@ -43,45 +44,43 @@ Update the code so it makes a `POST` request to the API endpoint. Then type your
 Your code should create a new `XMLHttpRequest`.
 
 ```js
-assert(code.match(/new\s+?XMLHttpRequest\(\s*?\)/g));
+assert.match(code, /new\s+?XMLHttpRequest\(\s*?\)/g);
 ```
 
 Your code should use the `open` method to initialize a `POST` request to the server.
 
 ```js
-assert(code.match(/\.open\(\s*?('|")POST\1\s*?,\s*?url\s*?,\s*?true\s*?\)/g));
+assert.match(code, /\.open\(\s*?('|")POST\1\s*?,\s*?url\s*?,\s*?true\s*?\)/g);
 ```
 
 Your code should use the `setRequestHeader` method.
 
 ```js
-assert(
-  code.match(
-    /\.setRequestHeader\(\s*?('|")Content-Type\1\s*?,\s*?('|")application\/json;\s*charset=UTF-8\2\s*?\)/g
-  )
+assert.match(
+  code,
+  /\.setRequestHeader\(\s*?('|")Content-Type\1\s*?,\s*?('|")application\/json;\s*charset=UTF-8\2\s*?\)/g
 );
 ```
 
 Your code should have an `onreadystatechange` event handler set to a function.
 
 ```js
-assert(code.match(/\.onreadystatechange\s*?=/g));
+assert.match(code, /\.onreadystatechange\s*?=/g);
 ```
 
 Your code should get the element with class `message` and change its `textContent` to `userName loves cats`
 
 ```js
-assert(
-  code.match(
-    /document\.getElementsByClassName\(\s*?('|")message\1\s*?\)\[0\]\.textContent\s*?=\s*?.+?\.userName\s*?\+\s*?.+?\.suffix/g
-  )
+assert.match(
+  code,
+  /document\.getElementsByClassName\(\s*?('|")message\1\s*?\)\[0\]\.textContent\s*?=\s*?.+?\.userName\s*?\+\s*?.+?\.suffix/g
```

Comment:
```suggestion
  /document\.getElementsByClassName\(\s*('|")message\1\s*\)\[0\]\.textContent\s*=\s*.*\.userName\s*\+\s*.*\.suffix/g
```

Reply:
with the `.` isn't it better to make the match lazy insted of greedy to avoid the regex trying to match too much?

Reply:
I'm a little confused about this one. I need to review the differences between lazy and greedy regex again.

Reply:
Sure, we can keep the wildcards lazy. Updated the suggestion.